pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/SegmentAssignment.java

```diff
@@ -60,9 +62,13 @@
    *
    * @param currentAssignment Current segment assignment of the table (map from segment name to instance state map)
    * @param instancePartitionsMap Map from type (OFFLINE|CONSUMING|COMPLETED) to instance partitions
+   * @param tierInstancePartitionsMap Map from tierName to instance partitions
+   * @param sortedTiers List of Tiers sorted as per priority
    * @param config Configuration for the rebalance
    * @return Rebalanced assignment for the segments
    */
   Map<String, Map<String, String>> rebalanceTable(Map<String, Map<String, String>> currentAssignment,
-      Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap, Configuration config);
+      Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap,
+      @Nullable Map<String, InstancePartitions> tierInstancePartitionsMap, @Nullable List<Tier> sortedTiers,
```

Comment:
(nit) Slightly more readable if we put `sortedTiers` in front of `tierInstancePartitionsMap` as `tierInstancePartitionsMap` is describing the `sortedTiers`?